src/Discord/Builders/Components/SelectMenu.php

```diff
@@ -81,10 +81,47 @@ class SelectMenu extends Component
 
     /**
      * Creates a new select menu.
+     * 
+     * @param string $customId The custom_id for this SelectMenu. If no $customID given, a generic $custom_id is set
      */
-    public function __construct()
+    public function __construct(string $customId = null)
     {
-        $this->custom_id = $this->generateUuid();
+        $this->custom_id = $customId ?? $this->generateUuid(); 
+    }
+
+    /**
+     * Creates a new select menu.
+     *
+     * @param string $customId The custom_id for this Selectmenu
+     * 
+     * @return self
+     */
+    public static function new($customId = null): self
```

Comment:
You are making duplicate of the `new` function instead of modifying existing one
https://github.com/discord-php/DiscordPHP/blob/2b91e2dae67e3c5541af01dc409e407607684872/src/Discord/Builders/Components/SelectMenu.php#L132